packages/flutter_tools/lib/src/html_utils.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:html/dom.dart';
+import 'package:html/parser.dart';
+
+import 'base/common.dart';
+
+/// Placeholder for base href
+const String kBaseHrefPlaceholder = r'$FLUTTER_BASE_HREF';
+
+class IndexHtml {
+  IndexHtml(this._content);
+
+  String get content => _content;
+  String _content;
+
+  Document _getDocument() => parse(_content);
+
+  /// Parses the base href from the index.html file.
+  String get baseHref {
```

Comment:
nit, i would make this a method since it does a lot of non-trivial work.

See: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods

Reply:
Makes sense!